src/gui/addnewtorrentdialog.cpp

```diff
@@ -546,6 +546,7 @@ void AddNewTorrentDialog::contentLayoutChanged(const int index)
     const auto contentLayout = ((index == 0)
                                 ? BitTorrent::detectContentLayout(m_torrentInfo.filePaths())
                                 : static_cast<BitTorrent::TorrentContentLayout>(index));
+    m_torrentParams.filePaths = m_torrentInfo.filePaths();
```

Comment:
This will discard all file name changes made before, so it is unacceptable.

I'm analyzing the problem... but it has a low priority.

Reply:
Good point. Modified it to avoid discarding file name changes made before. It should now ensure that the root folder had in fact been stripped before adding it back.